core/templatetags/irap_request.py

```diff
@@ -118,28 +117,23 @@ def file_field(
         "status": "not started",
         "errors": errors,
     }
-
-    if multi_field:
+    if not is_mandatory:
         needed = getattr(obj, file_field_prefix + "_needed")
 
-        has_error = (
-            get_field_errors(context, file_field_prefix + "_needed")
-            or get_field_errors(context, file_field_prefix + "_file")
-            or get_field_errors(context, file_field_prefix + "_link")
-        )
+    has_error = (
+        get_field_errors(context, file_field_prefix + "_needed")
+        or get_field_errors(context, file_field_prefix + "_file")
+        or get_field_errors(context, file_field_prefix + "_link")
+    )
 
+    if not is_mandatory:
         if needed:
             context["status"] = "in progress"
         else:
             context["status"] = "not started"
```

Comment:
If further up you define a default for `needed` of `False`, you can clean this up a lil:
```suggestion
    if needed:
        context["status"] = "in progress"
    else:
        context["status"] = "not started"
```